middleware.go

```diff
@@ -521,7 +521,9 @@ func saveResponseIntoFile(c *Client, res *Response) error {
 func getBodyCopy(r *Request) (*bytes.Buffer, error) {
 	// If r.bodyBuf present, return the copy
 	if r.bodyBuf != nil {
-		return bytes.NewBuffer(r.bodyBuf.Bytes()), nil
+		bodyCopy := &bytes.Buffer{}
+		bodyCopy.Write(r.bodyBuf.Bytes())
```

Comment:
there's an unhandled error here. Anyway, I think it's clearer if we use io.Copy here instead